File: eval-reviews/2026-09-12-chained/workflow-recovery/typescript/astra/r1/starter/main.ts
import { readFileSync } from "node:fs";
import { DomainError, parseWorkflow, Simulator } from "./workflow.ts";
import { LeasedSimulator, parseLeasedWorkflow } from "./leased.ts";

let response: unknown;
try {
  const request = JSON.parse(
    readFileSync(0, "utf8"),
    (_key: string, value: unknown, context?: { source?: string }) => {
      // Preserve the protocol's integer-token rule before JSON's numeric coercion loses it.
      if (
        typeof value === "number" &&
        context?.source &&
        !/^-?(0|[1-9][0-9]*)$/.test(context.source)
      )
        throw new DomainError("INVALID_INPUT");
      return value;
    },
  ) as { input?: unknown };
  response = {
    ok: true,
    result: request?.input !== null && typeof request?.input === "object" &&
      Object.hasOwn(request.input, "workers")
      ? new LeasedSimulator(parseLeasedWorkflow(request.input)).run()
      : new Simulator(parseWorkflow(request?.input)).run(),
  };
} catch (error) {
  if (error instanceof DomainError)
    response = { ok: false, error: { code: error.code } };
  else if (error instanceof SyntaxError)
    response = { ok: false, error: { code: "INVALID_INPUT" } };
  else throw error;
}
process.stdout.write(JSON.stringify(response) + "\n");

File: eval-reviews/2026-09-12-chained/workflow-recovery/python/astra/r1/starter/main.py
"""One-request JSON adapter; each request owns a fresh simulator."""
import json
import sys
from workflow import DomainError, Simulator, integer, object_fields, parse_workflow, require


def unique_object(pairs: list[tuple]) -> dict:
    result = {}
    for key, value in pairs:
        require(key not in result)
        result[key] = value
    return result


def main() -> None:
    try:
        request = object_fields(json.load(sys.stdin, object_pairs_hook=unique_object),
                                {"protocol_version", "task", "input"})
        integer(request["protocol_version"], 1, 1)
        require(request["task"] == "workflow-recovery")
        result = Simulator(parse_workflow(request["input"])).run()
        response = {"ok": True, "result": result}
    except DomainError as error:
        response = {"ok": False, "error": {"code": error.code}}
    except (ValueError, KeyError, TypeError):
        response = {"ok": False, "error": {"code": "INVALID_INPUT"}}
    print(json.dumps(response, separators=(",", ":")))


if __name__ == "__main__":
    main()
